test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightServiceDocumentSerializerTests.cs

```diff
@@ -112,7 +113,186 @@ public void WriteNullFunctionImportShouldThrow()
+
+            // Singleton (Title different from Name)
+
+            yield return new object[]
+            {
+                new ODataServiceDocument
+                {
+                    Singletons = singletons
+                },
+                "{\"value\":[" +
+                "{\"name\":\"Company\",\"title\":\"BusinessEntity\",\"kind\":\"Singleton\",\"url\":\"http://tempuri.org/Company\"}" +
+                "]}"
+            };
+
+            var functionImports = new List<ODataFunctionImportInfo>
+            {
+                new ODataFunctionImportInfo { Name = "GetOpenOrders", Url = new Uri("http://tempuri.org/GetOpenOrders") },
+                new ODataFunctionImportInfo { Name = "GetTop5Customers", Url = new Uri("http://tempuri.org/GetTop5Customers") }
+            };
+
+            // FunctionImport
+
+            yield return new object[]
+            {
+                new ODataServiceDocument
+                {
+                    FunctionImports = functionImports
+                },
+                "{\"value\":[" +
+                "{\"name\":\"GetOpenOrders\",\"kind\":\"FunctionImport\",\"url\":\"http://tempuri.org/GetOpenOrders\"}," +
+                "{\"name\":\"GetTop5Customers\",\"kind\":\"FunctionImport\",\"url\":\"http://tempuri.org/GetTop5Customers\"}" +
+                "]}"
+            };
+
+            // FunctionImport (Collection containing duplicates)
+
+            var duplicatedFunctionImports = new List<ODataFunctionImportInfo>(functionImports);
+            duplicatedFunctionImports[1] = duplicatedFunctionImports[0];
+
+            yield return new object[]
+            {
+                new ODataServiceDocument
+                {
+                    FunctionImports = duplicatedFunctionImports
+                },
+                "{\"value\":[" +
+                "{\"name\":\"GetOpenOrders\",\"kind\":\"FunctionImport\",\"url\":\"http://tempuri.org/GetOpenOrders\"}" +
+                "]}"
+            };
+
+
+            // Multiple element types
+
+            yield return new object[]
+            {
+                new ODataServiceDocument
+                {
+                    EntitySets = entitySets,
+                    Singletons = singletons,
+                    FunctionImports = functionImports
+                },
+                "{\"value\":[" +
+                "{\"name\":\"Customers\",\"kind\":\"EntitySet\",\"url\":\"http://tempuri.org/Customers\"}," +
+                "{\"name\":\"Orders\",\"kind\":\"EntitySet\",\"url\":\"http://tempuri.org/Orders\"}," +
+                "{\"name\":\"Company\",\"title\":\"BusinessEntity\",\"kind\":\"Singleton\",\"url\":\"http://tempuri.org/Company\"}," +
+                "{\"name\":\"GetOpenOrders\",\"kind\":\"FunctionImport\",\"url\":\"http://tempuri.org/GetOpenOrders\"}," +
+                "{\"name\":\"GetTop5Customers\",\"kind\":\"FunctionImport\",\"url\":\"http://tempuri.org/GetTop5Customers\"}" +
+                "]}"
+            };

```

Comment:
Are action imports also included in the function imports list?

Reply:
Yes